### ISSUE TITLE
JOSS pre-submission enquiry

### ISSUE THREAD

**Issue by greg-harhay** [openjournals/joss#394]
**Project repository URL:** INSERT YOUR REPOSITORY LINK HERE

Briefly describe your software, its research application and any questions you have about the JOSS submission criteria. 

I have 2 question

1) I am a US Government Scientist writing bioinformatics code & pipelines.  I have been instructed by my tech transfer people that I am not authorized to release code under an OSI license, but can under a Creative Commons Zero ([CC0](https://creativecommons.org/share-your-work/public-domain/cc0)) license.  Is this acceptable to JOSS?

2)  My pipeline involves using the proprietary app [Matlab](https://www.mathworks.com), the paid-for version of [Blast2GO](https://www.blast2go.com), and [Pathway Tools](http://bioinformatics.ai.sri.com/ptools/).  Since my pipeline involves a proprietary application, is this acceptable subject matter for JOSS?

**Comment by labarba** [comment]
JOSS only accepts submissions under an OSI-approved license (bear in mind we are an OSI affiliate!). So, no, you cannot use CC0 and submit to JOSS.

But you may have to go back to your tech transfer people, who may not be aware of the [Federal Source Code Policy's Pilot Program](https://code.gov/#/policy-guide/docs/open-source/licensing). In short, not only are they incorrect in saying that you must use CC0---the policy says in fact that "Your agency should choose a standard license," and points to OSI, specifically.

Works created by civil servants in the course of their official duties are _not protected by copyright_, according to 17 U.S. Code § 105,  https://www.law.cornell.edu/uscode/text/17/105 ---note that the law doesn't mention "public domain," but the end result is that government works, once released, become public domain in the U.S. This has been interpreted by some agencies to mean that software they release can only have a public-domain dedication, such as CC0.

Public domain dedication, to be clear, is not a software license. It does not ensure software freedom.

The Federal Source Code Policy’s Pilot Program clarified guidance for using open source licenses.

See also #179.

**Comment by greg-harhay** [comment]
Thank you for the clarification. I’ll pass this information on to my tech transfer people.

Best,
Greg

**Comment by apyle** [comment]
@labarba can you clarify the JOSS position for me? According to the [public domain question](https://opensource.org/faq#public-domain) on the OSI FAQ, "works authored by U.S. government employees, on government time and as part of their job, are automatically in the public domain," and not subject to copyright protection. Since these works cannot have a copyright they cannot have a license applied to them either according to the FAQ. 

OSI goes on further to state, "[t]here are certain circumstances, such as with U.S. government works as described above, where it is not easy to apply a license, and the software must be released into the public domain. In these cases, while it would be inaccurate to display the OSI logo or say that the license is OSI-approved (since there is no license), nevertheless we think it is accurate to say that such software is effectively open source, or open source for most practical purposes, even though it is not officially released under an open source license. (This is assuming, of course, that in the laws of releasing jurisdiction the meaning of "public domain" is compatible with the Open Source Definition.) After all, the freedoms guaranteed by open source licenses are still present, and it is possible for the familiar dynamics of open source collaboration to arise around the software."

If, in OSI's opinion, this code cannot have an OSI-approved license applied to it, and JOSS will not accept submissions unless it has an OSI-approved license, does this mean that JOSS will not accept any submissions from U.S. government employees? I want to make sure we are clear on that before encouraging U.S. government researchers to submit to JOSS.

**Comment by labarba** [comment]
**The JOSS policy is that an OSI-approved license is a requirement for consideration.**

IANAL, but here is what I know: 
1. works by civil servants are not protected by copyright, according to 17 U.S. Code § 105, https://www.law.cornell.edu/uscode/text/17/105 -- therefore, once released, they become public domain _in the US_ (i.e., this does not apply in other jurisdictions).
2. the Federal Source Code Policy’s Pilot Program clarified guidance for using open source licenses, and specifically recommends OSI licenses: https://code.gov/#/policy-guide/docs/open-source/licensing

It must be concluded that it's possible for software being public domain in the US, and attaching a license to say how code can be used, modified, redistributed.

Note that the text from the OSI website that you quoted above says that for U.S. government works, "it is not easy to apply a license." It does not say that it's impossible to do so. The Federal Source Code Policy relieved the difficulty, so the quoted text could be previous to that policy.

**Comment by apyle** [comment]
If the code is listed as public domain in the U.S. and has a OSI-approved license outside of the U.S. would JOSS then consider the submission?

**Comment by labarba** [comment]
But ... why?  The Federal Source Code policy says US Government agencies should use OSi licenses to release software.

@arfon might want to weigh in, at this point.

**Comment by apyle** [comment]
Yes - the FSCP recommends using OSI licenses and that works for the source code where the U.S. Government has procured adequate data rights as described in the [FSCP procurement page](https://code.gov/#/policy-guide/docs/compliance/procurement). It does not cover code where a copyright cannot be applied, such as source code produced by U.S. government employees as part of their official duties.

**Comment by arfon** [comment]
Thanks for the ping @labarba. I don't have anything further to add here I'm afraid.

> Yes - the FSCP recommends using OSI licenses and that works for the source code where the U.S. Government has procured adequate data rights as described in the FSCP procurement page. It does not cover code where a copyright cannot be applied, such as source code produced by U.S. government employees as part of their official duties.

This wasn't my understanding of the Federal Source Code Policy’s Pilot Program (https://code.gov/#/policy-guide/docs/open-source/licensing) and it seems like both @ianhbell (at NIST) https://github.com/openjournals/joss-reviews/issues/559#issuecomment-361643319 and @jhollist (at the EPA) https://github.com/openjournals/joss/issues/179#issuecomment-308898011 received guidance from their lawyers that seemed to not agree with this assessment @apyle.

As @labarba mentioned above, CC0 _alone_ is not acceptable to JOSS but a dual-licensed submission would be.

Sorry I can't add any further insights here.
